### PR TITLE
[MBL-897] Feature Flag Use of AI Project Tab

### PR DESCRIPTION
**Pull request by msadoon**
# 📲 What

Part of the new project page "Use of AI" tab.

# 🤔 Why

Feature flag is standard for launching new feature.

# 🛠 How

Added to `RemoteConfigFeatures`, `RemoteConfigFeature+Helpers`, `RemoteConfigFeatureFlagToolsViewModel`.

`use_of_ai_project_tab` also added to RemoteConfig on Firebase Remote Config. Productio, Beta and Alpha. 

# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# ✅ Acceptance criteria

- [x] Feature flag is available for use via a helper method.

### CODE REVIEW

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/RemoteConfig/RemoteConfigFeature+HelpersTests.swift:41:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/RemoteConfig/RemoteConfigFeature+HelpersTests.swift:50:1 | warning: (indent) Indent code in accordance with the scope level. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by scottkicks — CHANGES_REQUESTED** [review]
Why is `GraphAPI` being updated 🤔? This mutation should have been updated when I cleaned up this mutation in my TriggerThirdPartyEvent implementation. 

We need this update, but it would be better to address this in its own PR and keep the context of this one to the feature flag

File: Library/RemoteConfig/RemoteConfigFeature+HelpersTests.swift [hunk 1]
```diff
@@ -39,6 +39,15 @@ final class RemoteConfigFeatureHelpersTests: TestCase {
...
   }
 
+  func testCreatorDashboard_RemoteConfig_FeatureFlag_True() {
+    let mockRemoteConfigClient = MockRemoteConfigClient()
+      |> \.features .~ [RemoteConfigFeature.creatorDashboardEnabled.rawValue: true]
+
+    withEnvironment(remoteConfigClient: mockRemoteConfigClient) {
+      XCTAssertTrue(featureCreatorDashboardEnabled())
+    }
+  }
```
Comment (scottkicks) [inline Library/RemoteConfig/RemoteConfigFeature+HelpersTests.swift:49]: nice add

File: Library/RemoteConfig/RemoteConfigFeature+HelpersTests.swift [hunk 2]
```diff
@@ -47,4 +56,22 @@ final class RemoteConfigFeatureHelpersTests: TestCase {
       XCTAssertFalse(featureCreatorDashboardEnabled())
     }
   }
+
+  func testUseOfAIProjectAI_RemoteConfig_FeatureFlag_False() {
```
Comment (scottkicks) [inline Library/RemoteConfig/RemoteConfigFeature+HelpersTests.swift:60]:
```suggestion
  func testUseOfAIProjectTab_RemoteConfig_FeatureFlag_False() {
```

**Review by scottkicks — COMMENTED** [review]
We also should update the feature flag view controller's snapshot tests

**Comment by msadoon** [comment]
> We also should update the feature flag view controller's snapshot tests

Yes! Missed this, good catch.

**Review by scottkicks — APPROVED** [review]
👍 
just need to update with `main`

**Pull request merged**